.github/workflows/ci.yml

```diff
@@ -19,7 +19,7 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3
```

Comment:
@phisco can you help me understand the policy for updating these actions? is this done on every new commit of the actions? any reason they don't follow a semver approach instead so we only pick up actual releases?

Reply:
Sorry, missed this comment. This is to be sure we are referencing specific actions at every moment. It's debatable, and it's clearly leaning more toward security than usability as it can be a bit pedantic, we could think tuning it down a little bit and grouping them together at least 🤔 It takes the version that was originally specified, `actions/checkout@v3`, pins the the hash and move the version information in a comment, `actions/checkout@...hash... # v3`, and keeps the hash at the latest for the given version, `v3`.

Reply:
Grouping them all together in: https://github.com/crossplane/crossplane-runtime/pull/428